### PR TITLE
chore(docs): Add Agility CMS information

### PR DESCRIPTION
**Pull request by joelvarty**
## Description

Added an information page for AgilityCMS and a section in the CMS page about AgilityCMS.

### CODE REVIEW

File: docs/docs/sourcing-from-agilitycms.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+---
+title: Sourcing from Agility CMS
+---
+
+This guide takes you through the steps involved in setting up a Gatsby site that fetches content from [Agility CMS](https://agilitycms.com/).
+
+## What is Agility CMS? What makes it different?
+
+[Agility CMS](https://agilitycms.com/) is a headless Content Management System (CMS) that let's you easily define your custom content types and relationships easily.  This is called Content Architecture, and you can reuse this content for your websites and apps.
```
Comment (muescha) [inline docs/docs/sourcing-from-agilitycms.md]: > Avoid `easy` and `simple`
(source [Gatsby Style Guide](https://www.gatsbyjs.org/contributing/gatsby-style-guide/))

```suggestion
[Agility CMS](https://agilitycms.com/) is a headless Content Management System (CMS) that let's you define your custom content types and relationships.  This is called Content Architecture, and you can reuse this content for your websites and apps.
```

File: docs/docs/sourcing-from-agilitycms.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+In addition, Agility CMS provides a page routing API, which allows you to offload control of the sitemap to the content editors.
+
+All content is available through the Agility CMS Fetch or Preview API.
+
+
+## Getting Started
+
+### Create A Free Agility Account
+It only takes a minute to create an Agility CMS account, and it's free forever. [Sign up here](https://account.agilitycms.com/sign-up?product=agility-free).
```
Comment (muescha) [inline docs/docs/sourcing-from-agilitycms.md]: Avoid time so no one is frustrated if he takes longer times. 
Make more clear there many plans and not always free forever. 

```suggestion
Create an Agility CMS account with the Free Plan (this plan is free forever). [Sign up to Agility CMS](https://account.agilitycms.com/sign-up?product=agility-free).  
```

File: docs/docs/sourcing-from-agilitycms.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+### Create A Free Agility Account
+It only takes a minute to create an Agility CMS account, and it's free forever. [Sign up here](https://account.agilitycms.com/sign-up?product=agility-free).  
+
+Once your account is created, we'll need to grab your GUID and API Keys.
+
+### Get the Code
+
+Make sure you have Gatsby CLI tools installed (we are using npm here...)
+```
```
Comment (muescha) [inline docs/docs/sourcing-from-agilitycms.md]:
```suggestion
```shell
```

File: docs/docs/sourcing-from-agilitycms.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+npm install -g gatsby-cli
+```
+
+Go ahead and clone the Agility CMS Gatsby Starter repo on from GitHub that has all the code you need to get started.
+```
```
Comment (muescha) [inline docs/docs/sourcing-from-agilitycms.md]:
```suggestion
```shell
```

File: docs/docs/sourcing-from-agilitycms.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+git clone https://github.com/agility/agility-gatsby-starter.git
+```
+
+Resolve any dependencies
+
+```
```
Comment (muescha) [inline docs/docs/sourcing-from-agilitycms.md]:
```suggestion
```shell
```

File: docs/docs/sourcing-from-agilitycms.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+npm install
+```
+
+Run it in development mode!
+```
```
Comment (muescha) [inline docs/docs/sourcing-from-agilitycms.md]:
```suggestion
```shell
```

File: docs/docs/sourcing-from-agilitycms.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+gatsby develop
+```
+
+The site is just a starter, but it has a bunch of interesting features that you can use to build from.  Let's hook this code up to your new Agility CMS instance that you just created.
+
+## Hook it up to your Agility CMS instance
+Edit the **gatsby-config.js** file and repace the *guid* and *apiKey* with yours.
```
Comment (muescha) [inline docs/docs/sourcing-from-agilitycms.md]:
```suggestion
Edit the `gatsby-config.js` file and repace the `guid` and `apiKey` with yours.  
```

File: docs/docs/sourcing-from-agilitycms.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+```
+
+The site is just a starter, but it has a bunch of interesting features that you can use to build from.  Let's hook this code up to your new Agility CMS instance that you just created.
+
+## Hook it up to your Agility CMS instance
+Edit the **gatsby-config.js** file and repace the *guid* and *apiKey* with yours.  
+
+You can find your API keys on the Getting Started page in the Agility Content Manager.
+
+![alt text](https://help.agilitycms.com/hc/article_attachments/360037700311/Screen_Shot_2019-09-04_at_9.12.15_AM.png "Agility CMS - Dashboard - API Keys")
```
Comment (muescha) [inline docs/docs/sourcing-from-agilitycms.md]: Should we avoid external media and include the screenshot to Gatsby?

File: docs/docs/sourcing-from-agilitycms.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+
+
+If you use the "preview" key, you won't have to publish to see the changes you've made show up. If you use the "fetch" key, make sure you've published any content you wish to see changed.
```
Comment (muescha) [inline docs/docs/sourcing-from-agilitycms.md]:
```suggestion
If you use the `preview` key, you won't have to publish to see the changes you've made show up. If you use the `fetch` key, make sure you've published any content you wish to see changed.
```

File: docs/docs/sourcing-from-agilitycms.md [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+
+## How Does It Work
+
+The Gatsby Source Plugin downloads all the Pages on the Agility CMS Sitemap, as well as any Shared Content that's referenced on the *sharedContent* property on of the gatsby-config.js file.
```
Comment (muescha) [inline docs/docs/sourcing-from-agilitycms.md]:
```suggestion
The Gatsby Source Plugin downloads all the Pages on the Agility CMS Sitemap, as well as any Shared Content that's referenced on the `sharedContent` property on of the `gatsby-config.js` file.
```

File: docs/docs/sourcing-from-agilitycms.md [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+
+All of those pages and content are then made available in GraphQL to the React Components you will write to render those pages.
+
+Check out the Jumbotron component, which is used to render the Jumbotron module.  Here it is in the Agility Content Manager:
+
+![Alt Text](https://res.cloudinary.com/practicaldev/image/fetch/s--OQKoH_Nv--/c_limit%2Cf_auto%2Cfl_progressive%2Cq_auto%2Cw_880/https://thepracticaldev.s3.amazonaws.com/i/mjv2brybc17iyqdoy25u.png "Agility CMS - Example Module - Jumbotron")
```
Comment (muescha) [inline docs/docs/sourcing-from-agilitycms.md]: Avoid external media. Include to gatsby

File: docs/docs/sourcing-from-agilitycms.md [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+
+
+And here is the code used to render it.  Notice that the *title* and *subTitle* fields are available as properties the *item.fields* object.
```
Comment (muescha) [inline docs/docs/sourcing-from-agilitycms.md]:
```suggestion
And here is the code used to render it.  Notice that the `title` and `subTitle` fields are available as properties the `item.fields` object.
```

File: docs/docs/sourcing-from-agilitycms.md [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+
+``` javascript
```
Comment (muescha) [inline docs/docs/sourcing-from-agilitycms.md]: Should add also a filename as title
```suggestion
```javascript:title=???
```

File: docs/docs/sourcing-from-agilitycms.md [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+            <section className="jumbotron">
+                <h1>{this.props.item.fields.title}</h1>
+                <h2>{this.props.item.fields.subTitle}</h2>
+            </section>
+        );
+    }
+}
+```
+
+When we add new modules and content definitions to Agility, the components we use to render those modules will automatically get the strongly typed data delivered to those modules as props.
```
Comment (muescha) [inline docs/docs/sourcing-from-agilitycms.md]:
```suggestion
When we add new modules and content definitions to Agility CMS, the components we use to render those modules will automatically get the strongly typed data delivered to those modules as props.  
```

File: docs/docs/sourcing-from-agilitycms.md [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+
+If you use the "preview" key, you won't have to publish to see the changes you've made show up. If you use the "fetch" key, make sure you've published any content you wish to see changed.
+
+## How Does It Work
+
+The Gatsby Source Plugin downloads all the Pages on the Agility CMS Sitemap, as well as any Shared Content that's referenced on the *sharedContent* property on of the gatsby-config.js file.
+
+All of those pages and content are then made available in GraphQL to the React Components you will write to render those pages.
+
+Check out the Jumbotron component, which is used to render the Jumbotron module.  Here it is in the Agility Content Manager:
```
Comment (muescha) [inline docs/docs/sourcing-from-agilitycms.md]: Can you describe what is `Jumbotron module`? Is it a special component or what this means? The reader of this guide is not aware about special namings of Agility CMS at this point.

**Comment by muescha** [comment]
I miss also an entry in the `sites.yaml`

File: docs/docs/sourcing-from-agilitycms.md [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+Once your account is created, we'll need to grab your GUID and API Keys.
+
+### Get the Code
+
+Make sure you have Gatsby CLI tools installed (we are using npm here...)
+```
+npm install -g gatsby-cli
+```
+
+Go ahead and clone the Agility CMS Gatsby Starter repo on from GitHub that has all the code you need to get started.
```
Comment (muescha) [inline docs/docs/sourcing-from-agilitycms.md]:
```suggestion
Go ahead and clone the [Agility CMS Gatsby Starter](https://github.com/agility/agility-gatsby-starter) repo on from GitHub that has all the code you need to get started.
```

File: docs/docs/sourcing-from-agilitycms.md (hunk 7 above)
Comment (muescha) [reply docs/docs/sourcing-from-agilitycms.md]: Are the keys public or should we suggest using secret environment variables? (Also then in readme of the source-plugin)

File: docs/docs/sourcing-from-agilitycms.md [hunk 17]
```diff
@@ -0,0 +1,89 @@
...
+```
+gatsby develop
+```
+
+The site is just a starter, but it has a bunch of interesting features that you can use to build from.  Let's hook this code up to your new Agility CMS instance that you just created.
+
+## Hook it up to your Agility CMS instance
+Edit the **gatsby-config.js** file and repace the *guid* and *apiKey* with yours.  
+
+You can find your API keys on the Getting Started page in the Agility Content Manager.
```
Comment (muescha) [inline docs/docs/sourcing-from-agilitycms.md]:
```suggestion
You can find your API keys on the Getting Started page in the Agility CMS Content Manager.
```

File: docs/docs/sourcing-from-agilitycms.md (hunk 15 above)
Comment (joelvarty) [reply docs/docs/sourcing-from-agilitycms.md]: I've updated this text to be more descriptive.

**Comment by joelvarty** [comment]
I've made some changes based on feedback (thanks @muescha !) I haven't added Agility CMS to the "www/src/data/sidebars/doc-links.yaml" file because it instructed me not to do so in the comments of that file under the headless cms section.

File: docs/docs/sourcing-from-agilitycms.md [hunk 18]
```diff
@@ -0,0 +1,89 @@
...
+You can find your API keys on the Getting Started page in the Agility CMS Content Manager.
+
+
+![Agility CMS - Dashboard - API Keys](./images/agilitycms-api-keys.png)
+
+If you use the `preview` key, you won't have to publish to see the changes you've made show up. If you use the `fetch` key, make sure you've published any content you wish to see changed.
+
+## How Does It Work
+
+The Gatsby Source Plugin downloads all the Pages on the Agility CMS Sitemap, as well as any Shared Content that's referenced on the `sharedContent` property on of the `gatsby-config.js` file.
```
Comment (marcysutton) [inline docs/docs/sourcing-from-agilitycms.md]:
```suggestion
The Gatsby Source Plugin downloads all the Pages on the Agility CMS Sitemap, as well as any Shared Content that's referenced on the `sharedContent` property in the `gatsby-config.js` file.
```

File: docs/docs/sourcing-from-agilitycms.md [hunk 19]
```diff
@@ -0,0 +1,89 @@
...
+
+All of those pages and content are then made available in GraphQL to the React Components you will write to render those pages.
+
+Check out the component called "Jumbotron".  This is a simple example of how to display a styled heading and sub-heading with content that comes from Agility CMS.  Here is the Module that provides this content being edited in the Agility CMS Content Manager:
```
Comment (marcysutton) [inline docs/docs/sourcing-from-agilitycms.md]:
```suggestion
Check out the component called "Jumbotron".  This is an example of how to display a styled heading and sub-heading with content that comes from Agility CMS.  Here is the Module that provides this content being edited in the Agility CMS Content Manager:
```

File: docs/docs/sourcing-from-agilitycms.md [hunk 20]
```diff
@@ -0,0 +1,89 @@
...
+
+![Agility CMS - Example Module - Jumbotron](./images/agilitycms-jumbotron.png)
+
+
+And here is the code used to render it.  Notice that the `title` and `subTitle` fields are available as properties the `item.fields` object.
```
Comment (marcysutton) [inline docs/docs/sourcing-from-agilitycms.md]:
```suggestion
And here is the code used to render it.  Notice that the `title` and `subTitle` fields are available as properties of the `item.fields` object.
```

File: docs/docs/sourcing-from-agilitycms.md [hunk 21]
```diff
@@ -0,0 +1,89 @@
...
+            <section className="jumbotron">
+                <h1>{this.props.item.fields.title}</h1>
+                <h2>{this.props.item.fields.subTitle}</h2>
+            </section>
+        );
+    }
+}
+```
+
+When we add new modules and content definitions to Agility CMS, the components we use to render those modules will automatically get the strongly typed data delivered to those modules as props.
```
Comment (marcysutton) [inline docs/docs/sourcing-from-agilitycms.md]:
```suggestion
When you add new modules and content definitions to Agility CMS, the components used to render those modules will automatically receive the strongly typed data delivered to those modules as props.  
```

A conclusion with where to go for further information would be good to wrap up this doc!

File: docs/docs/sourcing-from-agilitycms.md [hunk 22]
```diff
@@ -0,0 +1,89 @@
...
+Edit the `gatsby-config.js` file and repace the `guid` and `apiKey` with yours.  
+
+You can find your API keys on the Getting Started page in the Agility CMS Content Manager.
+
+
+![Agility CMS - Dashboard - API Keys](./images/agilitycms-api-keys.png)
+
+If you use the `preview` key, you won't have to publish to see the changes you've made show up. If you use the `fetch` key, make sure you've published any content you wish to see changed.
+
+## How Does It Work
```
Comment (marcysutton) [inline docs/docs/sourcing-from-agilitycms.md]:
```suggestion
## How it works
```

File: docs/docs/sourcing-from-agilitycms.md [hunk 23]
```diff
@@ -0,0 +1,89 @@
...
+
+
+## Getting Started
+
+### Create A Free Agility Account
+Create an Agility CMS account with the Free Plan (this plan is free forever). [Sign up to Agility CMS](https://account.agilitycms.com/sign-up?product=agility-free).  
+
+Once your account is created, we'll need to grab your GUID and API Keys.
+
+### Get the Code
```
Comment (marcysutton) [inline docs/docs/sourcing-from-agilitycms.md]:
```suggestion
### Get the code
```

File: docs/docs/sourcing-from-agilitycms.md [hunk 24]
```diff
@@ -0,0 +1,89 @@
...
+[Agility CMS](https://agilitycms.com/) is a headless Content Management System (CMS) that let's you define your custom content types and relationships.  This is called Content Architecture, and you can reuse this content for your websites and apps.
+
+In addition, Agility CMS provides a page routing API, which allows you to offload control of the sitemap to the content editors.
+
+All content is available through the Agility CMS Fetch or Preview API.
+
+
+## Getting Started
+
+### Create A Free Agility Account
```
Comment (marcysutton) [inline docs/docs/sourcing-from-agilitycms.md]:
```suggestion
### Create a free Agility account
```

File: docs/docs/sourcing-from-agilitycms.md [hunk 25]
```diff
@@ -0,0 +1,89 @@
...
+## What is Agility CMS? What makes it different?
+
+[Agility CMS](https://agilitycms.com/) is a headless Content Management System (CMS) that let's you define your custom content types and relationships.  This is called Content Architecture, and you can reuse this content for your websites and apps.
+
+In addition, Agility CMS provides a page routing API, which allows you to offload control of the sitemap to the content editors.
+
+All content is available through the Agility CMS Fetch or Preview API.
+
+
+## Getting Started
```
Comment (marcysutton) [inline docs/docs/sourcing-from-agilitycms.md]:
```suggestion
## Getting started
```
Updating to sentence case headings to match the Gatsby Style Guide

File: docs/docs/sourcing-from-agilitycms.md (hunk 24 above)
Comment (muescha) [reply docs/docs/sourcing-from-agilitycms.md]:
```suggestion
### Create a free Agility CMS account
```

**Comment by joelvarty** [comment]
Thanks, @marcysutton! I've committed your suggestions.

File: docs/docs/sourcing-from-agilitycms.md [hunk 26]
```diff
@@ -0,0 +1,87 @@
...
+All of those pages and content are then made available in GraphQL to the React Components you will write to render those pages.
+
+Check out the component called "Jumbotron".  This is an example of how to display a styled heading and sub-heading with content that comes from Agility CMS.  Here is the Module that provides this content being edited in the Agility CMS Content Manager:
+
+![Agility CMS - Example Module - Jumbotron](./images/agilitycms-jumbotron.png)
+
+
+And here is the code used to render it.  Notice that the `title` and `subTitle` fields are available as properties of the `item.fields` object.
+
+```javascript:title=/src/modules/Jumbotron.js
```
Comment (LekoArts) [inline docs/docs/sourcing-from-agilitycms.md]:
```suggestion
```javascript:title=src/modules/Jumbotron.js
```

Small nitpick - other docs don't have this leading slash :)

**Comment by joelvarty** [comment]
Thank @LekoArts - suggestion committed!

**Comment by joelvarty** [comment]
Hi there @LekoArts @marcysutton @muescha  - anything else need to happen to get this merged?

**Pull request merged**